FreeRTOS-Plus/Source/FreeRTOS-IoT-Libraries-LTS-Beta2/c_sdk/platform/freertos/mbedtls/mbedtls_platform_freertos.c

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/**
+ * @file mbedtls_platform.c
+ * @brief Implements mbed TLS platform functions for FreeRTOS.
+ */
+
+/* FreeRTOS includes. */
+#include "FreeRTOS.h"
+#include "FreeRTOS_Sockets.h"
+
+/* mbed TLS includes. */
+#include "mbedtls_config.h"
+#include "threading_alt.h"
+#include "mbedtls/entropy.h"
+
+/*-----------------------------------------------------------*/
+
+void * mbedtls_platform_calloc( size_t nmemb,
+                                size_t size )
+{
+    size_t totalSize = nmemb * size;
+    void * pBuffer = NULL;
+
+    /* Check that neither nmemb nor size were 0. */
+    if( totalSize > 0 )
+    {
+        /* Overflow check. */
+        if( totalSize / size == nmemb )
+        {
+            pBuffer = pvPortMalloc( totalSize );
+
+            if( pBuffer != NULL )
+            {
+                ( void ) memset( pBuffer, 0x00, totalSize );
+            }
+        }
+    }
+
+    return pBuffer;
+}
+
+/*-----------------------------------------------------------*/
+
+void mbedtls_platform_free( void * ptr )
+{
+    vPortFree( ptr );
+}
+
+/*-----------------------------------------------------------*/
+
+int mbedtls_platform_send( void * ctx,
+                           const unsigned char * buf,
+                           size_t len )
+{
+    Socket_t socket = ctx;
```

Comment:
Does the compiler accept the typecast without any warnings?

Reply:
No, I will recheck that. I think it is better to add the cast.